### PR TITLE
[Android][TASK][58885] Splash + Home Screen

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Feature 58885: Splash + Home Screen](https://edu-redmine.sun-asterisk.vn/issues/58885)
## WHAT
- Splash Screen
- Home screen
## Evidence (Screenshot or Video)
sc (1)

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

**Review by hungld-1302 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by minhnn-1279 — APPROVED** [review]

File: app/src/main/java/vn/ztech/software/movie_streaming/ui/home/HomeFragment.kt [hunk 1]
```diff
@@ -31,7 +56,20 @@ class HomeFragment() : BaseFragment<FragmentHomeBinding>(FragmentHomeBinding::in
 
     override fun observeData() {
         viewModel.trendingMovies.observe(this){
-            binding?.tv?.text = it.toString()
+            it.results?.let {
+                listMoviesTrendingAdapter.submitData(it)
+                showTop1TrendingMovie(it.first() as Movie)
+
```
Comment (daolq-2712) [inline app/src/main/java/vn/ztech/software/movie_streaming/ui/home/HomeFragment.kt]:
```suggestion
```
remove blank line!

File: app/src/main/java/vn/ztech/software/movie_streaming/data/model/Show.kt [hunk 2]
```diff
@@ -1,11 +1,6 @@
...
-    val image: String?,
-    val latestEpisode: String?,
-    val season: String?,
-    val title: String?,
-    val type: String?,
-    val url: String?
-)
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
```
Comment (daolq-2712) [inline app/src/main/java/vn/ztech/software/movie_streaming/data/model/Show.kt]:
```suggestion
```
remove a blank line!

File: app/src/main/java/vn/ztech/software/movie_streaming/data/repository/MovieRepositoryImpl.kt [hunk 3]
```diff
@@ -5,6 +5,9 @@ import vn.ztech.software.movie_streaming.base.BaseRepository
 import vn.ztech.software.movie_streaming.data.model.*
 import vn.ztech.software.movie_streaming.data.source.IMovieDataSource
 import vn.ztech.software.movie_streaming.utils.DataResult
+import vn.ztech.software.movie_streaming.data.model.Media.Movie
+import vn.ztech.software.movie_streaming.data.model.Media.Show
+
```
Comment (daolq-2712) [inline app/src/main/java/vn/ztech/software/movie_streaming/data/repository/MovieRepositoryImpl.kt:11]:
```suggestion
```
remove a blank line!

File: app/src/main/java/vn/ztech/software/movie_streaming/ui/home/HomeFragment.kt [hunk 4]
```diff
@@ -1,15 +1,25 @@
...
 import vn.ztech.software.movie_streaming.base.BaseFragment
 import vn.ztech.software.movie_streaming.databinding.FragmentHomeBinding
 import org.koin.androidx.viewmodel.ext.android.viewModel
 import vn.ztech.software.movie_streaming.R
+import vn.ztech.software.movie_streaming.data.model.Media
+import vn.ztech.software.movie_streaming.data.model.Media.Movie
+import vn.ztech.software.movie_streaming.ui.moviedetails.MediaDetailsFragment
+import vn.ztech.software.movie_streaming.ui.search.SearchFragment
+import vn.ztech.software.movie_streaming.utils.extensions.loadImage
 import vn.ztech.software.movie_streaming.utils.extensions.showAlertDialog
```
Comment (daolq-2712) [inline app/src/main/java/vn/ztech/software/movie_streaming/ui/home/HomeFragment.kt]:
```suggestion
```
remove a blank line!

File: app/src/main/java/vn/ztech/software/movie_streaming/ui/home/HomeFragment.kt [hunk 5]
```diff
@@ -31,7 +56,20 @@ class HomeFragment() : BaseFragment<FragmentHomeBinding>(FragmentHomeBinding::in
 
     override fun observeData() {
         viewModel.trendingMovies.observe(this){
-            binding?.tv?.text = it.toString()
+            it.results?.let {
+                listMoviesTrendingAdapter.submitData(it)
+                showTop1TrendingMovie(it.first() as Movie)
+
+            }
+
```
Comment (daolq-2712) [inline app/src/main/java/vn/ztech/software/movie_streaming/ui/home/HomeFragment.kt]:
```suggestion
```
remove blank line!

File: app/src/main/java/vn/ztech/software/movie_streaming/ui/moviedetails/MediaDetailsFragment.kt [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import vn.ztech.software.movie_streaming.data.model.Media
+import vn.ztech.software.movie_streaming.data.model.Media.Movie
+import vn.ztech.software.movie_streaming.databinding.FragmentMovieDetailsBinding
+
+class MediaDetailsFragment() : BaseFragment<FragmentMovieDetailsBinding>(FragmentMovieDetailsBinding::inflate) {
+
+    override val viewModel: MovieDetailsViewModel by viewModel()
+
+    override fun initView() {
+        //todo implement later
```
Comment (daolq-2712) [inline app/src/main/java/vn/ztech/software/movie_streaming/ui/moviedetails/MediaDetailsFragment.kt]:
```suggestion
        //TODO implement later
```

File: app/src/main/java/vn/ztech/software/movie_streaming/ui/moviedetails/MediaDetailsFragment.kt [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    override fun initData() {
+        arguments?.takeIf { it.containsKey(BUNDLE_MOVIE) }?.let {
+            viewModel.media.value = it.getParcelable<Movie>(BUNDLE_MOVIE)
+        }
+    }
+
+    override fun bindData() {
+        //todo implement later
```
Comment (daolq-2712) [inline app/src/main/java/vn/ztech/software/movie_streaming/ui/moviedetails/MediaDetailsFragment.kt]:
```suggestion
        //TODO implement later
```

File: app/src/main/res/layout/fragment_movie_details.xml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="#f22"
```
Comment (daolq-2712) [inline app/src/main/res/layout/fragment_movie_details.xml]:
```suggestion
    android:background="#f22"
```
hard code!

File: app/src/main/res/layout/fragment_movie_details.xml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    tools:context=".ui.home.HomeFragment">
+
+    <!-- TODO: Update blank fragment layout -->
+    <TextView
+        android:id="@+id/tv"
```
Comment (daolq-2712) [inline app/src/main/res/layout/fragment_movie_details.xml]:
```suggestion
        android:id="@+id/tv"
```
https://github.com/framgia/coding-standards/blob/master/vn/android/codingstyleguide.md#321-%C4%91%E1%BA%B7t-t%C3%AAn-id

File: app/src/main/res/layout/fragment_movie_details.xml [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:text="Profile" />
```
Comment (daolq-2712) [inline app/src/main/res/layout/fragment_movie_details.xml]: .

Comment (daolq-2712) [reply app/src/main/res/layout/fragment_movie_details.xml]: Có thể sử dụng ``tools:text`` & hard code nếu chỉ dùng để test design

File: app/src/main/res/layout/image_view_blur.xml [hunk 11]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    xmlns:tools="http://schemas.android.com/tools"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+    <ImageView
+        android:id="@+id/iv_cover_blur"
```
Comment (daolq-2712) [inline app/src/main/res/layout/image_view_blur.xml]: https://github.com/framgia/coding-standards/blob/master/vn/android/codingstyleguide.md#321-%C4%91%E1%BA%B7t-t%C3%AAn-id

**Review by daolq-2712 — CHANGES_REQUESTED** [review]
Pls check my comments!

**Review by daolq-2712 — APPROVED** [review]
OK!

**Pull request merged**